### PR TITLE
feat: HTTP based Slash Commands util

### PR DESCRIPTION
**Pull request by DjDeveloperr**
Adds utility method for HTTP-based Slash Commands, and also helper methods for Opine and Oak framework.
Along with that, updated types for Member Screening feature. And added SlashBuilder, GuildTextChannel#bulkDelete and TextChannel#fetchMessages.

### CODE REVIEW

File: src/types/slash.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export interface SlashCommandChoice {
   /** (Display) name of the Choice */
   name: string
   /** Actual value to be sent in Interaction */
-  value: string
+  value: any
```
Comment (AkiaCode) [inline src/types/slash.ts:53]: any?

**Review by AkiaCode — APPROVED** [review]

File: src/models/slashClient.ts [hunk 2]
```diff
@@ -401,4 +460,78 @@ export class SlashClient {
...
+    next: CallableFunction
+  ): Promise<any> {
+    const verified = await this.verifyOpineRequest(req)
+    if (!verified) return res.setStatus(401).end()
+
+    await next()
+    return true
+  }
+
+  // TODO: create verifyOakMiddleware too
```
Comment (AkiaCode) [inline src/models/slashClient.ts:515]: i think there need draft

File: src/types/slash.ts (hunk 1 above)
Comment (DjDeveloperr) [reply src/types/slash.ts:53]: Value depends on the Command Option type.

File: src/models/slashClient.ts (hunk 2 above)
Comment (DjDeveloperr) [reply src/models/slashClient.ts:515]: No, it's just an idea to add it. I'm not good with Oak so.

File: src/types/slash.ts (hunk 1 above)
Comment (AkiaCode) [reply src/types/slash.ts:53]: Okay

File: src/models/slashClient.ts (hunk 2 above)
Comment (AkiaCode) [reply src/models/slashClient.ts:515]: > No, it's just an idea to add it. I'm not good with Oak so.

Ok

File: src/structures/textChannel.ts [hunk 3]
```diff
@@ -185,4 +229,40 @@ export class GuildTextChannel extends TextChannel {
...
+        .filter(
+          (e) =>
+            new Date().getTime() - e.createdAt.getTime() <=
+            1000 * 60 * 60 * 24 * 14
+        )
+        .map((e) => e.id)
+    }
+
+    ids = [...new Set(ids)]
+    if (ids.length < 2 || ids.length > 100)
```
Comment (Helloyunho) [inline src/structures/textChannel.ts:259]: Later we might can make multiple delete requests so it can delete more than 100 messages

**Review by Helloyunho — APPROVED** [review]
Seems not bad, I couldn't do a good review on `slashClient` because it was kinda hard for me to understand 😢

File: src/structures/textChannel.ts (hunk 3 above)
Comment (DjDeveloperr) [reply src/structures/textChannel.ts:259]: Wouldn't it be better to let user handle that logic?

Comment (Helloyunho) [reply src/structures/textChannel.ts:259]: I don't think so because many beginner developers make that mistake many times

Comment (DjDeveloperr) [reply src/structures/textChannel.ts:259]: Hmm, can we limit it to an additional option then?

Comment (Helloyunho) [reply src/structures/textChannel.ts:259]: yeah that's good too

**Pull request merged**